jgrapht-core/src/main/java/org/jgrapht/intervalgraph/IntervalGraph.java

```diff
@@ -1,14 +1,468 @@
+
+            specifics.removeVertex(v);// remove the vertex itself
+
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Set<V> vertexSet()
+    {
+        if (unmodifiableVertexSet == null) {
+            unmodifiableVertexSet = Collections.unmodifiableSet(specifics.getVertexSet());
+        }
+
+        return unmodifiableVertexSet;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public double getEdgeWeight(E e)
+    {
+        if (e == null) {
+            throw new NullPointerException();
+        }
+        return intrusiveEdgesSpecifics.getEdgeWeight(e);
+    }
+
+    /**
+     * Set an edge weight.
+     *
+     * @param e the edge
+     * @param weight the weight
+     * @throws UnsupportedOperationException if the graph is not weighted
+     */
+    @Override
+    public void setEdgeWeight(E e, double weight)
+    {
+        if (e == null) {
+            throw new NullPointerException();
+        }
+        intrusiveEdgesSpecifics.setEdgeWeight(e, weight);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public GraphType getType()
+    {
+        //TODO
+        if (directed) {
+            return new DefaultGraphType.Builder()
+                    .directed().weighted(weighted).allowMultipleEdges(allowingMultipleEdges)
+                    .allowSelfLoops(allowingLoops).build();
+        } else {
+            return new DefaultGraphType.Builder()
+                    .undirected().weighted(weighted).allowMultipleEdges(allowingMultipleEdges)
+                    .allowSelfLoops(allowingLoops).build();
+        }
+    }
+
+    /**
+     * Create the specifics for this graph. Subclasses can override this method in order to adjust
+     * the specifics and thus the space-time tradeoffs of the graph implementation.
+     *
+     * @param directed if true the specifics should adjust the behavior to a directed graph
+     *        otherwise undirected
+     * @return the specifics used by this graph
+     */
+    protected IntervalSpecifics<V, E> createSpecifics(boolean directed)
+    {
+        return new IntervalSpecifics<>(this);
+    }
+
+    /**
+     * Create the specifics for the edges set of the graph.
+     *
+     * @param weighted if true the specifics should support weighted edges
+     * @return the specifics used for the edge set of this graph
+     */
+    protected IntrusiveEdgesSpecifics<V, E> createIntrusiveEdgesSpecifics(boolean weighted)
+    {
+        if (weighted) {
+            return new WeightedIntrusiveEdgesSpecifics<>();
+        } else {
+            return new UniformIntrusiveEdgesSpecifics<>();
+        }
+    }
+
+    /**
+     * Adds edges between <code>sourceVertex</code> and every vertex from <code>vertices</code>
+     * This method should only be called from IntervalSpecifics, because interval graphs have by intervals defined edges.
```

Comment:
then consider using package private to enforce it syntactically would be a better choice. The behavior of invoker is unpredictable.

Reply:
I cannot do this, as the calling class lies in a different package. If you know a better way, without disturbing the whole structure of the library, tell me, please.